deploy/openshift/templates/che-server-template.yaml

```diff
@@ -0,0 +1,279 @@
+# Copyright (c) 2012-2018 Red Hat, Inc
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+---
+kind: Template
+apiVersion: v1
+metadata:
+  name: che
+  annotations:
+    description: Che
+objects:
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    labels:
+      app: che
+    name: che
+- apiVersion: v1
+  kind: RoleBinding
+  metadata:
+    name: che
+  roleRef:
+    name: admin
+  subjects:
+  - kind: ServiceAccount
+    name: che
+- apiVersion: v1
+  kind: Service
+  metadata:
+    labels:
+      app: che
+    name: che-host
+  spec:
+    ports:
+    - name: http
+      port: 8080
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      app: che
+- apiVersion: v1
+  kind: Route
+  metadata:
+    labels:
+      app: che
+    name: che
+  spec:
+    to:
+      kind: Service
+      name: che-host
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    labels:
+      app: che
+    name: che
+  spec:
+    replicas: 1
+    revisionHistoryLimit: 2
+    selector:
+      app: che
+    strategy:
+      type: ${STRATEGY}
+    template:
+      metadata:
+        labels:
+          app: che
+      spec:
+        containers:
+        - env:
+          - name: NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: CHE_HOST
+            value: che-${NAMESPACE}.${ROUTING_SUFFIX}
+          - name: CHE_PORT
+            value: '8080'
+          - name: CHE_API
+            value: "${PROTOCOL}://che-${NAMESPACE}.${ROUTING_SUFFIX}/api"
+          - name: CHE_WEBSOCKET_ENDPOINT
+            value: "${WS_PROTOCOL}://che-${NAMESPACE}.${ROUTING_SUFFIX}/api/websocket"
+          - name: CHE_DEBUG_SERVER
+            value: 'false'
+          - name: CHE_INFRASTRUCTURE_ACTIVE
+            value: openshift
+          - name: CHE_INFRA_KUBERNETES_BOOTSTRAPPER_BINARY__URL
+            value: "${PROTOCOL}://che-${NAMESPACE}.${ROUTING_SUFFIX}/agent-binaries/linux_amd64/bootstrapper/bootstrapper"
+          - name: CHE_INFRA_KUBERNETES_MACHINE__START__TIMEOUT__MIN
+            value: '5'
+          - name: CHE_INFRA_KUBERNETES_MASTER__URL
+          - name: CHE_INFRA_KUBERNETES_OAUTH__TOKEN
+            value: "${TOKEN}"
+          - name: CHE_INFRA_KUBERNETES_PASSWORD
+            value: "${OPENSHIFT_PASSWORD}"
+          - name: CHE_INFRA_OPENSHIFT_PROJECT
+            value: "${CHE_INFRA_OPENSHIFT_PROJECT}"
+          - name: CHE_INFRA_KUBERNETES_PVC_STRATEGY
+            value: "${CHE_INFRA_KUBERNETES_PVC_STRATEGY}"
+          - name: CHE_INFRA_KUBERNETES_PVC_PRECREATE__SUBPATHS
+            value: 'true'
```

Comment:
Please add a variable here. It makes sense to configure it. On newest OpenShift instances it may be disabled, and workspace start may speed up.